spec/state_of_the_nation_spec.rb

```diff
@@ -245,6 +245,97 @@ class Country < ActiveRecord::Base
     end
   end
 
+  context ".active_in_period" do
+    let(:bounded_president) { country.presidents.create!(entered_office_at: day(1), left_office_at: day(4)) }
+    let(:unbounded_president) { country.presidents.create!(entered_office_at: day(4), left_office_at: nil) }
+    let(:president_with_empty_active_period) { country.presidents.create!(entered_office_at: day(4), left_office_at: day(4)) }
+
+    it "returns true for records that are active in the interval" do
+      expect(bounded_president).to be_active_in_interval(day(3), day(7))
+      expect(bounded_president).to be_active_in_interval(day(3), nil)
+      expect(bounded_president).to be_active_in_interval(nil, day(3))
+      expect(bounded_president).to be_active_in_interval(nil, nil)
+
+      expect(unbounded_president).to be_active_in_interval(day(4), day(12))
+      expect(unbounded_president).to be_active_in_interval(day(4), nil)
+      expect(unbounded_president).to be_active_in_interval(nil, day(12))
+      expect(unbounded_president).to be_active_in_interval(nil, nil)
+      expect(unbounded_president).to be_active_in_interval(day(4), day(4))
+    end
+
+    it "returns true for records with an empty activation period in the range" do
+      expect(president_with_empty_active_period).to be_active_in_interval(day(3), day(5))
+      expect(president_with_empty_active_period).to be_active_in_interval(day(3), nil)
+      expect(president_with_empty_active_period).to be_active_in_interval(nil, day(5))
+      expect(president_with_empty_active_period).to be_active_in_interval(nil, nil)
```

Comment:
I don't think this is correct. A record with an empty interval is **never** active.

This comes down to the fact that we allow `ignore_empty: false`, which in this case results in returning a record that can never be active from a method that starts with `active_`. I'm thinking that as part of that major version bump @waheedel suggested, we should also do away with the ignore empty option and just always ignore empty records.